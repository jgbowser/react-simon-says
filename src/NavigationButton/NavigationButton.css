.btn_container {
  width: fit-content;
  margin: 20px auto;
  background-color: green;
  padding: 25px 50px;
  border-radius: 10px;
}

.link_button {
  text-decoration: none;
  color: white;
  font-weight: 800;
  letter-spacing: 2px;
}