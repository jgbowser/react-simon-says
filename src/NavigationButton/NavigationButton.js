import React from 'react'
import { Link } from 'react-router-dom'
import './NavigationButton.css'

export default class NavigationButton extends React.Component {
  render() {
    return (
      <div className="btn_container">
        <Link to={'/play'} className="link_button">
          {this.props.message}
        </Link>
      </div>
    )
  }
}