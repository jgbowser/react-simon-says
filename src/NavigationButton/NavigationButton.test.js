import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'
import NavigationButton from './NavigationButton'

describe('NavigationButton Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<BrowserRouter><NavigationButton message="click me" /></BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })
  it('should display the corresponding string that is passed as a prop', () => {
    const tree = renderer
      .create(<BrowserRouter><NavigationButton message="Start Game" /></BrowserRouter>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})