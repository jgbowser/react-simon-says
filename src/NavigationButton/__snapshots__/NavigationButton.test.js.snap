// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationButton Component should display the corresponding string that is passed as a prop 1`] = `
<div
  className="btn_container"
>
  <a
    className="link_button"
    href="/play"
    onClick={[Function]}
  >
    Start Game
  </a>
</div>
`;
